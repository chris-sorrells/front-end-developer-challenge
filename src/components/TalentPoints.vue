<template>
  <div class="talent-points">
    <div>
      <span class="used" v-text="pointsUsed" />&nbsp;/&nbsp;
      <span class="allowed" v-text="pointsAllowed" />
    </div>
    <div class="label">Points Spent</div>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
export default {
  computed: {
    ...mapGetters(["pointsUsed", "pointsAllowed"])
  }
};
</script>

<style lang="scss">
.talent-points {
  text-align: center;
  font-size: 125%;
  margin: auto 0;
  padding: 15px;
  border: 1px solid rgba(255, 255, 255, 0.15);
  background: rgba(0, 0, 0, 0.6);

  @media (max-width: 950px) {
    order: -1;
    flex: 1 0 85%;
    margin-bottom: 25px;
  }

  .label {
    color: rgb(100, 150, 200);
  }
}
</style>
