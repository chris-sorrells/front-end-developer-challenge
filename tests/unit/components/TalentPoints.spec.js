import { shallowMount, createLocalVue } from "@vue/test-utils";
import TalentPoints from "@/components/TalentPoints.vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("TalentPoints.vue", () => {
  let getters;
  let store;

  beforeEach(() => {
    getters = {
      pointsUsed: () => 3,
      pointsAllowed: () => 6
    };

    store = new Vuex.Store({
      getters
    });
  });
  it("renders", () => {
    const wrapper = shallowMount(TalentPoints, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  });
});
