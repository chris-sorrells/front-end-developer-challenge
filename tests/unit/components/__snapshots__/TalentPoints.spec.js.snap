// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalentPoints.vue renders 1`] = `
<div
  class="talent-points"
>
  <div>
    <span
      class="used"
    >
      3
    </span>
     / 
    
    <span
      class="allowed"
    >
      6
    </span>
  </div>
   
  <div
    class="label"
  >
    Points Spent
  </div>
</div>
`;
